ibm/service/vpc/data_source_ibm_is_backup_policies.go

```diff
@@ -0,0 +1,357 @@
+									},
+									"resource_type": &schema.Schema{
+										Type:        schema.TypeString,
+										Computed:    true,
+										Description: "The type of resource referenced.",
+									},
+								},
+							},
+						},
+						"resource_group": &schema.Schema{
+							Type:        schema.TypeList,
+							Computed:    true,
+							Description: "The resource group for this backup policy.",
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"href": &schema.Schema{
+										Type:        schema.TypeString,
+										Computed:    true,
+										Description: "The URL for this resource group.",
+									},
+									"id": &schema.Schema{
+										Type:        schema.TypeString,
+										Computed:    true,
+										Description: "The unique identifier for this resource group.",
+									},
+									"name": &schema.Schema{
+										Type:        schema.TypeString,
+										Computed:    true,
+										Description: "The user-defined name for this resource group.",
+									},
+								},
+							},
+						},
+						"resource_type": &schema.Schema{
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "The type of resource referenced.",
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func dataSourceIBMIsBackupPoliciesRead(context context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	sess, err := vpcClient(meta)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	start := ""
+	matchBackupPolicies := []vpcv1.BackupPolicy{}
+
+	var resourceGroup string
+	if v, ok := d.GetOk("resource_group"); ok {
+		resourceGroup = v.(string)
+	}
+
+	var name string
+	if v, ok := d.GetOk("name"); ok {
+		name = v.(string)
+	}
+
+	var tag string
+	if v, ok := d.GetOk("tag"); ok {
+		tag = v.(string)
+	}
+
+	for {
+		listBackupPoliciesOptions := &vpcv1.ListBackupPoliciesOptions{}
+		if start != "" {
+			listBackupPoliciesOptions.Start = &start
+		}
+		if resourceGroup != "" {
+			listBackupPoliciesOptions.SetResourceGroupID(resourceGroup)
+		}
+		if name != "" {
+			listBackupPoliciesOptions.SetName(name)
+		}
+		if tag != "" {
+			listBackupPoliciesOptions.SetTag(tag)
+		}
+		backupPolicyCollection, response, err := sess.ListBackupPoliciesWithContext(context, listBackupPoliciesOptions)
+		if err != nil {
+			log.Printf("[DEBUG] ListBackupPoliciesWithContext failed %s\n%s", err, response)
+			return diag.FromErr(fmt.Errorf("[ERROR] ListBackupPoliciesWithContext failed %s\n%s", err, response))
+		}
+		if backupPolicyCollection != nil && *backupPolicyCollection.TotalCount == int64(0) {
+			break
+		}
+		start = flex.GetNext(backupPolicyCollection.Next)
+		matchBackupPolicies = append(matchBackupPolicies, backupPolicyCollection.BackupPolicies...)
+		if start == "" {
+			break
+		}
+
+	}
+	if len(matchBackupPolicies) == 0 {
+		return diag.FromErr(fmt.Errorf("[ERROR] no BackupPolicies found"))
```

Comment:
common naming convention and a proper messge BackupPolicies and backup_policies for all error messages